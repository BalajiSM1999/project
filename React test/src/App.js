import MainPage from './components/main';
import './style/App.css';

function App() {
  return (
    <div className="App">
  <MainPage />
    </div>
  );
}

export default App;
