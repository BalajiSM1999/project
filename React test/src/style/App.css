
.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
body{
  margin: 10px;
}
.card{
  border: 1px solid black;
  padding: 5px;
  width: 48%;
  margin: 2px;
  float: left;
}

.dropbtn {
  padding: 10px;
  background-color: #1e5564;
  color: white;
  border-radius: 3px;
  border: none;
  font-size: 12px;
  width: 140%;
}

.dropdown {
  position: relative;
  display: inline-block;
  margin-bottom: 15px;
  margin-top: 15px;
  margin-left: 750px;
}
.funcs{
  display: inline-flex;
  margin: 30px;
}

#searchTerm{
  padding: 7px;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: #f1f1f1;
  min-width: 120px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 1;
  cursor: pointer;
}

.dropdown-content a {
  color: black;
  padding: 5px;
  text-decoration: none;
  display: block;
}

.dropdown-content a:hover {background-color: #ddd;}

.dropdown:hover .dropdown-content {display: block;}

.dropdown:hover .dropbtn {background-color: #27839c;}

.charts{
  width: 500px;
  height: 500px;
  margin-left: 100px;
  border: 1px solid black;
  padding-bottom: 25px;
  padding: 35px;
  padding-left: 200px;
  padding-right: 200px;
}
.btn-primary{
  padding: 10px;
  background-color: #1e5564;
  color: white;
  border-radius: 3px;
  border: none;
  font-size: 12px;
  margin-left: 25px;
  margin-bottom: 5px;

}
.card-img{
display: flex;  
float: right;
width: 20%;
}
.heading{
  text-align: center;
}
.card-date{display: none;}