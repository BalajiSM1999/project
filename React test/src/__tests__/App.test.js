import App from "../App"
import renderer from "react-test-renderer"
import MainPage from "../components/main"
import {fireEvent, render } from "@testing-library/react"

it("renders main app",()=>{
    render(<App />)
})

it ("matches snapshot", ()=>{
    const tree= renderer.create(<App />);
    expect(tree).toMatchSnapshot();
    })

    it('should render drop button', () => {
        const {queryByTitle} =render(<MainPage/>);
        const btn = queryByTitle("drop-btn");
        expect(btn).toBeTruthy()
      });

      test("should be able to view charts", () =>{
        const mockFn = jest.fn();
        const {getByTestId} =render(<MainPage handleSubmit={mockFn} />);
        const button = getByTestId("btn");
        fireEvent.submit(button);
    
        expect(mockFn).toHaveBeenCalledTimes(0);
    })