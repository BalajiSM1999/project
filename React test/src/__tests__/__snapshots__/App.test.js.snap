// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <div>
    <button
      className="btn-primary"
      data-testid="btn"
      onClick={[Function]}
    >
      View Charts
    </button>
    
    <br />
    <div
      class="dropdown"
    >
      <button
        class="dropbtn"
        title="drop-btn"
      >
        Sort By
      </button>
      <div
        class="dropdown-content"
        id="sort-content"
      >
        <a
          onClick={[Function]}
        >
          Name
        </a>
        <a
          onClick={[Function]}
        >
          Price
        </a>
        <a
          onClick={[Function]}
        >
          Category
        </a>
      </div>
    </div>
    <p>
      Search: 
      <input
        id="searchTerm"
        onKeyPress={[Function]}
        type="text"
      />
    </p>
    <div
      id="divided"
    />
  </div>
</div>
`;
